Automattic/jetpack
# Self-Heal IDC when home / siteurl align to WP.com values

Jetpack Identity Crisis are usually triggered when multiple websites are attempting to send data to WP.com using a token that was connected under a different domain. Recently it was identified that if the sync query args are modified the site can get into an Identity Crisis with itself. This patch adds an additional check so that if the site has the same domain as connected on WP.com the IDC option is self-cleared.



Fixes https://wp.me/p9dueE-1Nw

#### Changes proposed in this Pull Request:

* Updates validate_sync_error_idc_option to check the current values against WP.com expected values. if they match it is not a valid IDC conflict.

#### Jetpack product discussion
* Bug Fix

#### Does this pull request change what data or activity we track or use?
* No

#### Testing instructions:

##### Verify valid IDC issues persist

* Duplicate your Jetpack Test Site (Site A) to a new Test Site (Site B). (**Copy over the entire DB but have home/site_url options be a new domain)
* Login to Test Site B
* Edit a Post
* Verify site is placed into Staging Mode
* Login to Test Site A
* Edit a Post
* Verify site is not placed into Staging Mode

##### Verify invalid IDC issues self-heal

* Apply the following patch to your mu-plugins or similar file.
```
function local_sync_query_args( $query_args ) {
$query_args['home'] = 'mdbitz.fire';
$query_args['site_url'] = 'mdbitz.fire';
return $query_args;
}
add_filter( 'jetpack_sync_send_data_query_args', 'local_sync_query_args', 100, 1 );
```
* Login to the Test Site
* Edit a Post
* Verify the site is not placed into Staging Mode
* Verify sync action is sent to WP.com successfully.

#### Proposed changelog entry for your changes:
* Bug Fix - Update Identity Crisis validation to resolve conflicts where sites were flagged as staging against themselves.
